From 044829ddb5fc053f0d4dcc847ea1988f89317917 Mon Sep 17 00:00:00 2001
From: "Wulf C. Krueger" <philantrop@exherbo.org>
Date: Wed, 19 Feb 2014 17:58:15 +0100
Subject: [PATCH] build: man/local.mk: Don't regenerate the man pages

The help2man step loves to fail if done during cross-compiles and for
chroots. Don't try rebuilding the man pages which we pre-generate and
put into a separate tarball. Preparing the man pages tarball works like
this:
1. cave resolve coreutils -z1x --abort-at-phase install
2. From ${WORK}, take the .1 files in the man sub-directory
3. Add all .1 files to coreutils-${PV}-man-pages.tar.xz, preserving the
paths, e. g. coreutils-8.22/man.
4. Upload it to dev.exherbo.org:/srv/www/dev.exherbo.org/distfiles/
---
 man/local.mk | 110 -----------------------------------------------------------
 1 file changed, 110 deletions(-)

diff --git a/man/local.mk b/man/local.mk
index 7ce426b..8d25814 100644
--- a/man/local.mk
+++ b/man/local.mk
@@ -40,7 +40,6 @@ CLEANFILES += $(ALL_MANS)
 # install a binary, require that all programs be built at distribution
 # time.  We can't use 'dist-hook' for this, since it would run too late:
 # the manpages must be generated before the distdir is created and filled.
-$(EXTRA_MANS): $(all_programs)
 
 # This is a kludge to remove generated 'man/*.1' from a non-srcdir build.
 # Without this, "make distcheck" might fail.
@@ -59,115 +58,6 @@ mandeps += $(top_srcdir)/src/system.h
 
 $(ALL_MANS): $(mandeps)
 
-# Most prog.1 man pages depend on src/prog.  List the exceptions:
-# Note that dir and vdir are exceptions only if you consider the name
-# of the .c file upon which they depend: ls.c.
-man/arch.1:      src/uname
-man/dir.1:       src/dir
-man/install.1:   src/ginstall
-man/vdir.1:      src/vdir
-
-man/base64.1:    src/base64
-man/basename.1:  src/basename
-man/cat.1:       src/cat
-man/chcon.1:     src/chcon
-man/chgrp.1:     src/chgrp
-man/chmod.1:     src/chmod
-man/chown.1:     src/chown
-man/chroot.1:    src/chroot
-man/cksum.1:     src/cksum
-man/comm.1:      src/comm
-man/cp.1:        src/cp
-man/csplit.1:    src/csplit
-man/cut.1:       src/cut
-man/date.1:      src/date
-man/dd.1:        src/dd
-man/df.1:        src/df
-man/dircolors.1: src/dircolors
-man/dirname.1:   src/dirname
-man/du.1:        src/du
-man/echo.1:      src/echo
-man/env.1:       src/env
-man/expand.1:    src/expand
-man/expr.1:      src/expr
-man/factor.1:    src/factor
-man/false.1:     src/false
-man/fmt.1:       src/fmt
-man/fold.1:      src/fold
-man/groups.1:    src/groups
-man/head.1:      src/head
-man/hostid.1:    src/hostid
-man/hostname.1:  src/hostname
-man/id.1:        src/id
-man/join.1:      src/join
-man/kill.1:      src/kill
-man/link.1:      src/link
-man/ln.1:        src/ln
-man/logname.1:   src/logname
-man/ls.1:        src/ls
-man/md5sum.1:    src/md5sum
-man/mkdir.1:     src/mkdir
-man/mkfifo.1:    src/mkfifo
-man/mknod.1:     src/mknod
-man/mktemp.1:    src/mktemp
-man/mv.1:        src/mv
-man/nice.1:      src/nice
-man/nl.1:        src/nl
-man/nohup.1:     src/nohup
-man/nproc.1:     src/nproc
-man/numfmt.1:    src/numfmt
-man/od.1:        src/od
-man/paste.1:     src/paste
-man/pathchk.1:   src/pathchk
-man/pinky.1:     src/pinky
-man/pr.1:        src/pr
-man/printenv.1:  src/printenv
-man/printf.1:    src/printf
-man/ptx.1:       src/ptx
-man/pwd.1:       src/pwd
-man/readlink.1:  src/readlink
-man/realpath.1:  src/realpath
-man/rm.1:        src/rm
-man/rmdir.1:     src/rmdir
-man/runcon.1:    src/runcon
-man/seq.1:       src/seq
-man/sha1sum.1:   src/sha1sum
-man/sha224sum.1: src/sha224sum
-man/sha256sum.1: src/sha256sum
-man/sha384sum.1: src/sha384sum
-man/sha512sum.1: src/sha512sum
-man/shred.1:     src/shred
-man/shuf.1:      src/shuf
-man/sleep.1:     src/sleep
-man/sort.1:      src/sort
-man/split.1:     src/split
-man/stat.1:      src/stat
-man/stdbuf.1:    src/stdbuf
-man/stty.1:      src/stty
-man/sum.1:       src/sum
-man/sync.1:      src/sync
-man/tac.1:       src/tac
-man/tail.1:      src/tail
-man/tee.1:       src/tee
-man/test.1:      src/test
-man/timeout.1:   src/timeout
-man/touch.1:     src/touch
-man/tr.1:        src/tr
-man/true.1:      src/true
-man/truncate.1:  src/truncate
-man/tsort.1:     src/tsort
-man/tty.1:       src/tty
-man/uname.1:     src/uname
-man/unexpand.1:  src/unexpand
-man/uniq.1:      src/uniq
-man/unlink.1:    src/unlink
-man/uptime.1:    src/uptime
-man/users.1:     src/users
-man/wc.1:        src/wc
-man/who.1:       src/who
-man/whoami.1:    src/whoami
-man/yes.1:       src/yes
-
 .x.1:
 	$(AM_V_GEN)name=`echo $@ | sed 's|.*/||; s|\.1$$||'` || exit 1;	\
 ## Ensure that help2man runs the 'src/ginstall' binary as 'install' when
-- 
1.9.0

